### DISCUSSION TITLE
What method can I use to judge whether all the files have been downloaded?

### DISCUSSION THREAD

**Discussion by bdc2018** [cefsharp/CefSharp#4188] [Q&A]
I'm going to use CefSharp to automatically download the pictures or other files of some web pages, and use OnResourceLoadComplete to save the files, but FrameLoadEnd can't judge whether all the files have been downloaded. What method can I use to judge whether all the files have been downloaded, and then switch to another web page to continue downloading?

**Comment by jankurianski** [comment]
Using JavaScript, poll the `document.readyState` until it becomes `complete`.
https://developer.mozilla.org/en-US/docs/Web/API/Document/readyState

`interactive` means "... images, stylesheets and frames are still loading"
`complete` means "all sub-resources have finished loading"

**Comment by amaitland** [comment]
> automatically download the pictures or other files of some web pages, and use OnResourceLoadComplete to save the files

You may wish to consider using a response filter see 
https://github.com/cefsharp/CefSharp/wiki/General-Usage#response-filtering
